tests/epyccel/test_epyccel_pyc_math.py

```diff
@@ -0,0 +1,132 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=wildcard-import
+import sys
+from numpy.random import randint, uniform
+from numpy import isclose
+from conftest       import *
+
+from pyccel.epyccel import epyccel
+from pyccel.decorators import types
+
+# -----------------------------------------------------------------------------
+@pytest.mark.parametrize( 'language', (
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="gcd not implemented in fortrand"),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_call_gcd(language):
+    @types(int, int)
+    def call_gcd(x, y):
+        from math import gcd
+        return gcd(x, y)
+
+    f = epyccel(call_gcd, language=language)
+    x = randint(1e9)
+    y = randint(1e9)
+
+    assert(f(x,y) == call_gcd(x, y))
+# -----------------------------------------------------------------------------
+@pytest.mark.parametrize( 'language', (
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="factorial not implemented in fortran"),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_call_factorial(language):
+    @types(int)
+    def call_factorial(x):
+        from math import factorial
+        return factorial(x)
+
+    f = epyccel(call_factorial, language=language)
+    x = randint(15)
+
+    assert(f(x) == call_factorial(x))
+# -----------------------------------------------------------------------------
+# New in version 3.9.
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="requires python3.9 or higher")
```

Comment:
Should we add github action testing for python 3.9?

Reply:
it's up to you guys. in my opinion, we need to add the features in the new python versions.

Reply:
@EmilyBourne Yes, I think it is time to remove version 3.5 and add version 3.9

Reply:
I have created PR #585 to fix this. Once this is merged these functions will be correctly tested